node/v2/handler.js

```diff
@@ -180,6 +181,14 @@ TChannelV2Handler.prototype.handleCallRequest = function handleCallRequest(reqFr
     if (self.remoteHostPort === null) {
         return callback(errors.TChannelCallReqBeforeInitReqError());
     }
+
+    if (!reqFrame.body ||
+        !reqFrame.body.headers ||
+        !reqFrame.body.headers.as
+    ) {
+        return callback(errors.AsHeaderRequired());
```

Comment:
Not callback(err), that's for low level "should close the socket" type errors.

In this case, this should be an operational error resulting in a `"BadRequest", "missing as header"` error frame to the caller.